### PR TITLE
done1

### PR DESCRIPTION
**Pull request by panianchuk**
https://panianchuk.github.io/react_dynamic-list-of-todos/

### CODE REVIEW

**Review by Volodymyr-Melnyk-Fron10der — APPROVED** [review]
Great job!

**Review by Volodymyr-Melnyk-Fron10der — CHANGES_REQUESTED** [review]
Sorry, I see some tests are failed, check it out please and then request new review

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,67 @@
...
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
```
Comment (DarkMistyRoom) [inline src/App.tsx:17]:
```suggestion
  const [isError, setIsError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
```

Comment (DarkMistyRoom) [reply src/App.tsx:17]: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

File: src/App.tsx [hunk 2]
```diff
@@ -17,18 +70,41 @@ export const App: React.FC = () => {
...
+                  <Loader />
+                ) : (
+                  <TodoList
+                    todos={visibleTodos}
+                    selectedTodo={selectedTodo}
+                    setSelectedTodo={setSelectedTodo}
+                  />
+                )}
+              </div>
+            )}
```
Comment (DarkMistyRoom) [inline src/App.tsx:97]: the nested ternary operator is evil. try to avoid it

File: src/components/TodoFilter/TodoFilter.tsx [hunk 3]
```diff
@@ -1,34 +1,63 @@
...
+            onChange={(e) => setTypeFilter(e.target.value)}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
```
Comment (DarkMistyRoom) [inline src/components/TodoFilter/TodoFilter.tsx:33]:
```suggestion
      <p className="
        control 
        is-expanded 
        has-icons-left 
        has-icons-right
      ">
```

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]

**Pull request closed without merging**